### PR TITLE
Linux & BSD support

### PR DESCRIPTION
**Pull request by Be-ing**
This is #4 squashed and with a few small commits split out, plus updating README.md, formatting changes, and fixing a few clippy warnings.

### CODE REVIEW

**Comment by Be-ing** [comment]
Unfortunately it seems the KDE method is not really working. I suggest we change to the same [method](https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/merge_requests/52/diffs) as KDE's implementation of the XDG Desktop Portal by reading the `BackgroundNormal` key under `Colors:Window` in `$XDG_CONFIG_DIR/kdeglobals` and calculating the grayscale value of that color.

**Comment by edfloreshz** [comment]
Alright, I'll get on that next 👍

**Comment by Be-ing** [comment]
Maybe [serde_ini](https://crates.io/crates/serde_ini) could help? I'm not sure.

**Comment by Be-ing** [comment]
The calculation of the grayscale value with [qGray](https://doc.qt.io/qt-5/qcolor.html#qGray) is simply `(r * 11 + g * 16 + b * 5)/32`.

**Comment by edfloreshz** [comment]
Done, could you test it before we merge @Be-ing?

File: src/freedesktop.rs [hunk 1]
```diff
@@ -34,36 +38,37 @@ fn get_freedesktop_color_scheme() -> Option<Mode> {
...
-        Ok(theme) => {
-            if theme.to_lowercase().contains("dark") {
-                Mode::Dark
-            } else {
-                Mode::Light
-            }
+fn detect_kde(path: &str) -> Mode {
+    match Config::new().file(path) {
+        Ok(cfg) => {
+            let rgb: Vec<u32> = match cfg.get_str("BackgroundNormal") {
```
Comment (Be-ing) [inline src/freedesktop.rs:51]: I don't think this `simple_config_parser` crate is capable of doing what's required here. Looking at my ~/.config/kdeglobals, there are several matches for `BackgroundNormal`. I think relevant one is the one under `Colors:Window`.

**Comment by frewsxcv** [comment]
I just enabled GitHub Actions for CI. Mentioning in case you want to merge/rebase to pick that up here

File: src/freedesktop.rs (hunk 1 above)
Comment (edfloreshz) [reply src/freedesktop.rs]: Replaced `simple_config_parser` with `rust-ini` 👍🏼

**Comment by edfloreshz** [comment]
After @Be-ing's approval, we'll be ready for merge @frewsxcv, should we merge or rebase?

**Comment by frewsxcv** [comment]
I have no preference between rebasing/merging/squashing. Do whatever you'd like!

File: src/lib.rs [hunk 2]
```diff
@@ -3,14 +3,12 @@
...
 //!
-//!     match mode {
-//!         dark_light::Mode::Dark => {},
-//!         dark_light::Mode::Light => {},
-//!     }
-//! }
+//!match mode {
+//!    dark_light::Mode::Dark => {},
+//!    dark_light::Mode::Light => {},
+//!}
```
Comment (frewsxcv) [inline src/lib.rs:11]: Small thing– Can you add one space after `//!` in each line?

Comment (edfloreshz) [reply src/lib.rs:11]: Sure thing, ok to merge after I fix this?

Comment (frewsxcv) [reply src/lib.rs:11]: Sounds good to me!

**Comment by edfloreshz** [comment]
Closes #1

**Comment by Be-ing** [comment]
I tested this on KDE Plasma 5.23.4 on Fedora 34 switching between several color schemes and it works!

**Comment by edfloreshz** [comment]
Awesome! Great job as well @Be-ing and @frewsxcv, cheers 🥂

**Comment by edfloreshz** [comment]
By the way, I've been improving the code, looks way better.

I'll try to upload the commit tonight so you can review it in the morning. unless you're a night owl 🦉

**Comment by Be-ing** [comment]
@frewsxcv shall we make a new release now that this is merged?

**Comment by edfloreshz** [comment]
How about we wait until we have #10 ready?

**Comment by Be-ing** [comment]
#10 may take some time to polish and implement on all platforms. I don't think we need to hold back a release for that. Version numbers are cheap. :)

**Comment by edfloreshz** [comment]
Alright then! 👍🏼

**Comment by frewsxcv** [comment]
0.2.0 released!

**Pull request merged**